segmentstore/storage/impl/src/test/java/io/pravega/segmentstore/storage/impl/filesystem/IdempotentStorageTest.java

```diff
@@ -0,0 +1,245 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.impl.filesystem;
+
+import io.pravega.segmentstore.contracts.BadOffsetException;
+import io.pravega.segmentstore.contracts.StreamSegmentNotExistsException;
+import io.pravega.segmentstore.storage.SegmentHandle;
+import io.pravega.segmentstore.storage.Storage;
+import io.pravega.segmentstore.storage.StorageTestBase;
+import io.pravega.test.common.AssertExtensions;
+import lombok.val;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.util.concurrent.CompletableFuture;
+
+import static io.pravega.test.common.AssertExtensions.assertMayThrow;
+import static io.pravega.test.common.AssertExtensions.assertThrows;
+
+/**
+ * Unit tests for FileSystemStorage.
+ */
+public abstract class IdempotentStorageTest extends StorageTestBase {
```

Comment:
A bunch of code from `FileSystemStorageTest` is being moved her. Is there any addition or change or is it simply code moving?